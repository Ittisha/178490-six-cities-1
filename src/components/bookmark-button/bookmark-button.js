import * as React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import {Operation} from '../../reducers/favorites/favorites';

class BookmarkButton extends React.PureComponent {
  constructor(props) {
    super(props);
    this._handleButtonClick = this._handleButtonClick.bind(this);
  }

  render() {
    const {
      className,
      isInBookmarks,
      width,
      height,
    } = this.props;
    const activeButtonClass = isInBookmarks ? `${className}__bookmark-button--active` : ``;
    const bookmarkHiddenText = isInBookmarks ? `In bookmarks` : `To bookmarks`;

    return <button
      className={`${className}__bookmark-button button ${activeButtonClass}`}
      type="button"
      onClick={this._handleButtonClick}
    >
      <svg
        className={`${className}__bookmark-icon`}
        width={width}
        height={height}
      >
        <use xlinkHref="#icon-bookmark"/>
      </svg>
      <span className="visually-hidden">{bookmarkHiddenText}</span>
    </button>;
  }

  _handleButtonClick() {
    const {
      isInBookmarks,
      offerId,
      updateFavorites,
    } = this.props;

    updateFavorites(Number(!isInBookmarks), offerId);
  }
}

BookmarkButton.propTypes = {
  offerId: PropTypes.number.isRequired,
  isInBookmarks: PropTypes.bool.isRequired,
  className: PropTypes.string.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  updateFavorites: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  updateFavorites: (isFavorite, id) => dispatch(Operation.updateFavorite(isFavorite, id))
});

export {BookmarkButton};
export default connect(null, mapDispatchToProps)(BookmarkButton);
