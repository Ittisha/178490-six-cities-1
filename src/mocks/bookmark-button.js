export const MOCK_PROPS = {
  offerId: 1,
  isInBookmarks: false,
  className: ``,
  width: 15,
  height: 12,
};
