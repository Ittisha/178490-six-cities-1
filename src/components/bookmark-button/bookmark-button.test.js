import * as React from 'react';
import renderer from 'react-test-renderer';
import {BookmarkButton} from './bookmark-button';

const MOCK_PROPS = {
  offerId: 1,
  isInBookmarks: false,
  className: ``,
  width: 15,
  height: 12,
};

it(`Bookmarks are rendered correctly`, () => {
  const tree = renderer
    .create(<BookmarkButton
      offerId={MOCK_PROPS.offerId}
      isInBookmarks={MOCK_PROPS.isInBookmarks}
      className={MOCK_PROPS.className}
      width={MOCK_PROPS.width}
      height={MOCK_PROPS.height}
      updateFavorites={() => {}}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});
