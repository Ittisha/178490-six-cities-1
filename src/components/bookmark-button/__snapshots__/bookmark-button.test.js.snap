// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmarks are rendered correctly 1`] = `
<button
  className="__bookmark-button button "
  onClick={[Function]}
  type="button"
>
  <svg
    className="__bookmark-icon"
    height={12}
    width={15}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
